src/rax.c

```diff
@@ -527,26 +527,30 @@ int raxGenericInsert(rax *rax, unsigned char *s, size_t len, void *data, void **
         /* Make space for the value pointer if needed. */
         if (!h->iskey || (h->isnull && overwrite)) {
             h = raxReallocForData(h,data);
-            if (h) memcpy(parentlink,&h,sizeof(h));
-        }
-        if (h == NULL) {
-            errno = ENOMEM;
-            return 0;
+            if (h == NULL) {
+                errno = ENOMEM;
+                return 0;
+            }
+            memcpy(parentlink,&h,sizeof(h));
         }
 
+        int res;
         /* Update the existing key if there is already one. */
         if (h->iskey) {
             if (old) *old = raxGetData(h);
             if (overwrite) raxSetData(h,data);
             errno = 0;
-            return 0; /* Element already exists. */
+            res = 0;    /* Element already exists. */
         }
 
-        /* Otherwise set the node as a key. Note that raxSetData()
-         * will set h->iskey. */
-        raxSetData(h,data);
-        rax->numele++;
-        return 1; /* Element inserted. */
+        /* A new element, note raxSetData() will set h->iskey. */
+        if (!h->iskey) {
+            raxSetData(h,data);
+            rax->numele++;
+            res = 1;    /* Element inserted. */
+        }
+
+        return res;
```

Comment:
I don't think this change makes it objectively better. It's just a change from one style to another. We avoid this kind of unnecessary changes.

Reply:
Okay, thanks for your reply. The code modification here is unnecessary, because the original code is already correct, it was just to increase the readability when reading, maybe adding some comments is a better way

Reply:
OK, I understand. I just don't think the change increases readability. The style you're most used to reading is usually easier to read, but it's differs from person to person.